### PR TITLE
feat: coinbase

### PR DESCRIPTION
**Pull request by Eikix**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #150 

## What is the new behavior?

- coinbase is now the execution_info.sequencer_address field
- it is up to the underlying full node to set this variable: https://github.com/keep-starknet-strange/madara/blob/6b369880ea7ebd574abf7860286fa6c93c973249/crates/pallets/starknet/src/lib.rs#L451

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/utils/src/traits.cairo [hunk 1]
```diff
@@ -1,3 +1,5 @@
+use core::option::OptionTrait;
```
Comment (enitrat) [inline crates/utils/src/traits.cairo:1]: no need

File: crates/utils/src/traits.cairo [hunk 2]
```diff
@@ -85,6 +87,13 @@ trait TryIntoResult<T, U> {
     fn try_into_result(self: T) -> Result<U, EVMError>;
 }
 
+impl EthAddressTryIntoResult of TryIntoResult<ContractAddress, EthAddress> {
```
Comment (enitrat) [inline crates/utils/src/traits.cairo:90]:
```suggestion
impl EthAddressTryIntoResultContractAddress of TryIntoResult<ContractAddress, EthAddress> {
```

**Review by enitrat — COMMENTED** [review]
So coinbase is going to return a 31bytes address right?

**Comment by Eikix** [comment]
> So coinbase is going to return a 31bytes address right?

More like, when we use Madara / Katana / any full node, we'll make sure to set the sequencer address to be a 20bytes address 

and record a corresponding sn address in our registry mappping

**Comment by enitrat** [comment]
> we'll make sure to set the sequencer address to be a 20bytes addres

How will you do that? It's going to take ages to compute

**Review by enitrat — DISMISSED** [review]

**Review by enitrat — APPROVED** [review]

**Pull request merged**